.github/workflows/ci.yaml

```diff
@@ -23,40 +23,20 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
 
-      - name: Cache pip dependencies
-        uses: actions/cache@v4
-        with:
-          path: ~/.cache/pip
-          key: pip-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/pyproject.toml') }}
-          restore-keys: |
-            pip-${{ runner.os }}-${{ matrix.python-version }}-
-
-      - name: Install Hatch
-        run: pip install --no-cache-dir hatch
-
-      - name: Cache Hatch environments
-        uses: actions/cache@v4
-        with:
-          path: ~/.local/share/hatch
-          key: hatch-env-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/pyproject.toml') }}
-          restore-keys: |
-            hatch-env-${{ runner.os }}-${{ matrix.python-version }}-
+      - name: Install uv
+        run: pip install uv
```

Comment:
nit: Consider replacing with the recommended setup here: https://docs.astral.sh/uv/guides/integration/github/#multiple-python-versions

I'm not sure if it would make a difference at all, but just in case you want to try it out.